File: tutorials/testing/python/graph_alist.py
from bridges.data_src_dependent import *
from bridges.graph_adj_list import *
from bridges.bridges import *
import sys


def main():

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # title, description
    bridges.set_title("A Simple Graph (Adjacency List) Example using IMDB Actor/Movie Data")
    bridges.set_description("Two lists each having an actor as the root node with 15 movies they played in as leaf nodes. "
            +	"Root nodes are both red, Kevin Bacon's leaf nodes are green and Denzel Washington's leaf nodes are neutral.")

    actor_movie_data = get_actor_movie_imdb_data(1813)

    # create an adjacency list based graph
    g = GraphAdjList()

    # first create vertices for two actors and add the to the graph
    a1 = "Kevin_Bacon_(I)"
    a2 = "Denzel_Washington"

    g.add_vertex(a1, "")
    g.add_vertex(a2, "")

    g.get_vertex(a1).color = "red"
    g.get_vertex(a1).size = 30
    g.get_vertex(a2).color = "red"
    g.get_vertex(a2).size = 30

    # add and edge between the two actors
    g.add_edge(a1, a2)
    g.get_link_visualizer(a1, a2).color = "magenta"
    g.get_link_visualizer(a1, a2).thickness = 4.0

    # get their nodes
    v1 = g.get_vertex(a1)
    v2 = g.get_vertex(a2)

    cnt1 = 0
    cnt2 = 0

    # we will find the first 15 immediate neighbors of the two actors
    # and color those links and nodes by following their adjacency lists
    for k in range(len(actor_movie_data)):
        # from the actor movie data, get and actor-movie pair
        a = actor_movie_data[k].actor
        m = actor_movie_data[k].movie

        if (a == "Kevin_Bacon_(I)" and cnt1 < 15):
            # add vertices for this movie and an edge for the link
            g.add_vertex(m, "")
            g.add_edge(a1, m)
            g.add_edge(m, a1)
            # make the movie node a bit transparent
            g.get_vertex(m).opacity = 0.5
            cnt1 += 1

        elif (a == "Denzel_Washington" and cnt2 < 15):
            # add vertices for this movie and an edge for the link
            g.add_vertex(m, "")
            g.add_edge(a2, m)
            g.add_edge(m, a2)
            # make the movie node a bit transparent
            g.get_vertex(m).opacity = 0.5
            cnt2 += 1

    # Next, we illustrate traversing the adjacency list and color the movie
    # nodes adjacent to the Kevin Bacon node

    # first get the adjacency list for Kevin Bacon
    for edge in g.out_going_edge_set_of(a1):
        f = edge.fromv
        to = edge.tov
        if to != a2:
            g.get_visualizer(to).color = "turquoise"

    for edge in g.out_going_edge_set_of(a2):
        f = edge.fromv
        to = edge.tov
        if to != a1:
            g.get_visualizer(to).color = "orange"

    # Pass the graph object to BRIDGES
    bridges.set_data_structure(g)
    # visualize the graph
    bridges.visualize()


if __name__ == "__main__":
    main()

File: tutorials/testing/java/graph_alist.java
import java.util.ArrayList;
import java.lang.String;
import bridges.base.Element;
import bridges.base.SLelement;
import bridges.base.GraphAdjListSimple;
import bridges.base.Edge;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.ActorMovieIMDB;

public class graph_alist {
	public static void main(String[] args) throws Exception {

		// initialize Bridges
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");

		// title, description
		bridges.setTitle("A Simple Graph (Adjacency List) Example using IMDB Actor/Movie Data");
		bridges.setDescription("Two lists each having an actor as the root node with 15 movies they played in as leaf nodes. "
			+	"Root nodes are both red, Kevin Bacon's leaf nodes are green and Denzel Washington's leaf nodes are neutral.");

		DataSource ds = bridges.getDataSource();
		ArrayList<ActorMovieIMDB>  actor_movie_data =
			(ArrayList<ActorMovieIMDB>) ds.getActorMovieIMDBData(1813);


		// create an adjacency list based graph
		GraphAdjListSimple<String> g = new GraphAdjListSimple<String>();

		// first create vertices for two actors and add them to the graph
		String kevin_bacon = "Kevin_Bacon_(I)", denzel_washington = "Denzel_Washington";
		g.addVertex(kevin_bacon, "");
		g.addVertex(denzel_washington, "");

		// color the two actor nodes
		g.getVertex(kevin_bacon).setColor("red");
		g.getVertex(kevin_bacon).setSize(30);
		g.getVertex(denzel_washington).setColor("red");
		g.getVertex(denzel_washington).setSize(30);

		// add an edge between the two actors kevin_bacon and denzel_washington,
		// the third parameter is the edge weight
		g.addEdge(kevin_bacon, denzel_washington);
		g.getLinkVisualizer(kevin_bacon, denzel_washington).setColor("magenta");
		g.getLinkVisualizer(kevin_bacon, denzel_washington).setThickness(4.0f);

		// get their nodes
		Element v1 = g.getVertex(kevin_bacon);
		Element v2 = g.getVertex(denzel_washington);

		// we will find the first 15 immediate neighbors of of the two actors
		// and color those links and nodes by followng their adjacency lists
		int cnt1 = 0, cnt2 = 0;
		for (int k = 0; k < actor_movie_data.size(); k++) {
			// from the actor movie data, get an actor-movie pair
			String a = actor_movie_data.get(k).getActor();
			String m = actor_movie_data.get(k).getMovie();

			if (a.equals(kevin_bacon) && cnt1 < 15) {

				// add vertices for this movie  and an edge for the link
				g.addVertex(m, "");
				g.addEdge(kevin_bacon, m);
				g.addEdge(m, kevin_bacon);

				// make the movie node a bit transparent
				g.getVertex(m).setOpacity(0.7f);
				cnt1++;
			}
			else if (a.equals(denzel_washington) && cnt2 < 15) {
				// add vertices for this movie  and an edge for the link
				g.addVertex(m, "");
				g.addEdge(denzel_washington, m);
				g.addEdge(m, denzel_washington);
				// make the movie node a bit transparent
				g.getVertex(m).setOpacity(0.7f);
				cnt2++;
			}
		}

		// Next, we illustrate traversing the adjacency list and color the
		// movie nodes adjacent to the Kevin Bacon node.

		// first get the adjacency list for Kevin Bacon
		// traverse the adjacency list
		for (Edge<String, String> edge : g.outgoingEdgeSetOf(kevin_bacon)) {
			String from = edge.getFrom(), to = edge.getTo();
			if (!to.equals(denzel_washington))
				edge.setColor("turquoise");
		}
		for (Edge<String, String> edge : g.outgoingEdgeSetOf(denzel_washington)) {
			String from = edge.getFrom(), to = edge.getTo();
			if (!to.equals(kevin_bacon))
				edge.setColor("orange");
		}

		// Pass the graph object to BRIDGES
		bridges.setDataStructure(g);

		// Finaly we call the visualize function
		bridges.visualize();
	}
}
